packages/orchestration/src/exos/chain-hub.js

```diff
@@ -389,32 +402,40 @@ export const makeChainHub = (zone, agoricNames, vowTools) => {
      * contract `start` the call will happen again naturally.
      *
      * @param {string} name
-     * @param {CosmosChainInfo} chainInfo
+     * @param {ChainInfo & { chainId: string }} chainInfo
      */
     registerChain(name, chainInfo) {
       chainInfos.init(name, chainInfo);
-      if (chainInfo.bech32Prefix) {
-        bech32PrefixToChainName.init(chainInfo.bech32Prefix, name);
+      if (/** @type {CosmosChainInfo} */ (chainInfo).bech32Prefix) {
+        bech32PrefixToChainName.init(
+          /** @type {CosmosChainInfo} */ (chainInfo).bech32Prefix,
+          name,
+        );
       }
     },
     /**
      * Update chain info by completely replacing existing entry
      *
      * @param {string} chainName - Name of the chain to update
-     * @param {CosmosChainInfo} chainInfo - New chain info
+     * @param {ChainInfo} chainInfo - New chain info
      * @throws {Error} If chain not registered
      */
     updateChain(chainName, chainInfo) {
       if (!chainInfos.has(chainName)) {
         throw makeError(`Chain ${q(chainName)} not registered`);
       }
       const oldInfo = chainInfos.get(chainName);
-      if (oldInfo.bech32Prefix) {
-        bech32PrefixToChainName.delete(oldInfo.bech32Prefix);
+      if (/** @type {CosmosChainInfo} */ (oldInfo).bech32Prefix) {
```

Comment:
chain-hub.js is ripe for `.ts` transformation. should be straightforward (though not of scope)